File: Models/AppConfig.cs
using System.Text.Json.Serialization;

namespace AnimeWallpaper.Models;

public class AppConfig
{
    [JsonPropertyName("wallhavenApiKey")]
    public string WallhavenApiKey { get; set; } = string.Empty;

    [JsonPropertyName("query")]
    public string Query { get; set; } = "anime";

    [JsonPropertyName("intervalMinutes")]
    public int IntervalMinutes { get; set; } = 30;

    [JsonPropertyName("minimumWidth")]
    public int MinimumWidth { get; set; } = 1920;

    [JsonPropertyName("minimumHeight")]
    public int MinimumHeight { get; set; } = 1080;

    [JsonPropertyName("sfwOnly")]
    public bool SfwOnly { get; set; } = true;

    [JsonPropertyName("random")]
    public bool Random { get; set; } = true;

    [JsonPropertyName("startWithWindows")]
    public bool StartWithWindows { get; set; } = true;

    [JsonPropertyName("maxCachedWallpapers")]
    public int MaxCachedWallpapers { get; set; } = 20;

    [JsonPropertyName("showNotifications")]
    public bool ShowNotifications { get; set; } = true;

    public AppConfig Clone()
    {
        return (AppConfig)MemberwiseClone();
    }
}

File: macos/Sources/Models/AppConfig.swift
import Foundation

public struct AppConfig: Codable, Equatable {
    public var wallhavenApiKey: String
    public var query: String
    public var intervalMinutes: Int
    public var minimumWidth: Int
    public var minimumHeight: Int
    public var sfwOnly: Bool
    public var random: Bool
    public var startWithMac: Bool
    public var maxCachedWallpapers: Int
    public var showNotifications: Bool

    public init(
        wallhavenApiKey: String = "",
        query: String = "anime",
        intervalMinutes: Int = 30,
        minimumWidth: Int = 1920,
        minimumHeight: Int = 1080,
        sfwOnly: Bool = true,
        random: Bool = true,
        startWithMac: Bool = false,
        maxCachedWallpapers: Int = 20,
        showNotifications: Bool = true
    ) {
        self.wallhavenApiKey = wallhavenApiKey
        self.query = query
        self.intervalMinutes = intervalMinutes
        self.minimumWidth = minimumWidth
        self.minimumHeight = minimumHeight
        self.sfwOnly = sfwOnly
        self.random = random
        self.startWithMac = startWithMac
        self.maxCachedWallpapers = maxCachedWallpapers
        self.showNotifications = showNotifications
    }
}
